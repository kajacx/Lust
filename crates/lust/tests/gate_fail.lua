-- @type string | number?
local value = nil

if value then
    -- @type number
    local valueInsideIf = value
end
