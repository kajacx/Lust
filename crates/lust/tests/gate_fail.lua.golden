tests/gate_fail.lua:0:0: Cannot assign value 'TODO: VarName("value")' of type 'Union(UnionType { variants: [String, Number] })' into the variable 'valueInsideIf' of type 'Number'.
